core/src/typecheck/subtyping.rs

```diff
@@ -0,0 +1,244 @@
+//! Type subsumption
+/// Type subsumption is generally used when we change from inference mode to checking mode.
+/// Currently, there is one subtyping relations :
```

Comment:
Here, and below, you should continue to use `//!` instead of `///`, because it's part of the top-level module documentation.